src/control/SelfAssessmentController.php

```diff
@@ -123,7 +123,10 @@ private function sendLinkViaEmail($submission, $emailAddress)
         $link = $submission->getLink();
 
         $email = new Email();
-        $email->setFrom($this->data()->ContactEmail);
+
+        if ($this->data()->ContactEmail) {
```

Comment:
This field is not validated as an email address, so using it for the setFrom may still trigger an error.
Either validate in SelfAssement.php, validate here or put a try catch around this whole email thing (or all 3 to be super safe)

Reply:
@alex-dna I have made those updates as suggested.